marlowe-playground-client/src/MainFrame.purs

```diff
@@ -140,6 +145,18 @@ toProjects halogen = do
     getState = view _projects
   (imapState setState getState <<< mapAction ProjectsAction) halogen
 
+toNewProject ::
+  forall m a.
+  Functor m =>
+  HalogenM NewProject.State NewProject.Action ChildSlots Void m a -> HalogenM FrontendState HAction ChildSlots Void m a
+toNewProject halogen = do
+  currentState <- get
+  let
+    setState = flip (set _newProject) currentState
+  let
+    getState = view _newProject
+  (imapState setState getState <<< mapAction NewProjectAction) halogen
```

Comment:
I think this is worth teasing out to the Marlowe-ish part and the Halogen-ish part. Something like:

``` purescript
mapSubmodule ::
  forall m output slots state state' action action'.
  Functor m =>
  Lens' state state' ->
  (action' -> action) ->
  HalogenM state' action' slots output m ~> HalogenM state action slots output m
mapSubmodule lens wrapper halogen = do
  state <- get
  let
    setState = flip (set lens) state

    getState = view lens
  imapState setState getState $ mapAction wrapper halogen


toNewProject = mapSubmodule _newProject NewProjectAction
```

...and then maybe moving `mapSubmodule` to somewhere like, erm, `Halogen.Extras` in `web-common`?

Reply:
nice idea moving it, done.